### DISCUSSION TITLE
camera is fixed above, long object moves below

### DISCUSSION THREAD

**Discussion by emard** [OpenStitching/stitching#81] [Q&A]
HI
I would like to automatically stitch 20-40 images
all of them are very planar and rectangular geometric
distorsion is negligible, optical details is slightly blurred
like camera is not fully in focus a bit (slight spherical aberration):

camera is fixed above, long object moves below

What happens with stiching, it only stitches maybe 5-6 from
40 pictures. I think it should at least report which pics are not
stiched and suggest to experiment with some settings

**Comment by lukasalexanderweber** [comment]
Hey,

I also thought about informing the user if not all of the images are stitched.

> I think it should at least report which pics are not
> stiched and suggest to experiment with some settings

this is a good idea. I will implement this if I find some time, or you can do it and create a PR if you wish

**Comment by lukasalexanderweber** [comment]
To summarize:

Mostly users of the stitching package will have a set of images which ALL should be included in the final panorama. By setting the confidence threshold to 1, by default images are sorted out if they don't have good enough matches. In this case the user should be informed.

**Comment by lukasalexanderweber** [comment]
BTW do you need help for your images or have you figured out how to generate a complete panorama?

**Comment by emard** [comment]
Of course I'd like some help with images
I can prepare some minimal set of images
that demonstrates the problem. Each image
is full HD 1920x1080 jpg, they overlap at least
30% left to right. Stitching about 40 of them
should produce about 40000x1080 image.

It's great if you can suggest solution I'd like to have
automatic script that would stitch such images
images are every time similar in color and content.

**Comment by lukasalexanderweber** [comment]
Yeah a Sample Set would be helpful

**Comment by emard** [comment]
HI

Here is 25MB zip download link valid in next 10 days there are
pictures that I'd like to stitch automatically
with commandline I used and result when not all pictures are in.

I'd like to stich them all horizontally left-to-right, sorted filename order.
Camera optics is pretty good x/y grid even and rectangular, same distance
from camera to object, perpendicular to surface and no need for any fisheye
lens correction or similar.

images from this equipment will be a bit dark, slightly unsharp and with
similar texture. If I need to overlap more, I can make more pictures with
less camera moving step.

https://jumboiskon.tportal.hr/download/1e80f803-29f8-4ebf-ae12-c1e24241075a

**Comment by lukasalexanderweber** [comment]
try

`stitch Screen* --confidence_threshold 0.2 --matches_graph_dot_file matches.txt  --detector sift`

and add 

`--final_megapix 0.01`

if you don't want to wait for so long if you're just trying out some settings.

`--matches_graph_dot_file matches.txt`

this will create a file with matching informations, especially the confidences. We can see that with the default settings the confidences are too low, thats why so many images are filtered out. We can lower the `--confidence_threshold` to include more images. Changing the Feature Detector to SIFT improves the matches significantly. Here is the resulting matching graph:

```
graph matches_graph{
"Screen20221018_132345.jpg" -- "Screen20221018_132408.jpg"[label="Nm=8, Ni=8, C=0.769231"];
"Screen20221018_132408.jpg" -- "Screen20221018_132445.jpg"[label="Nm=14, Ni=14, C=1.14754"];
"Screen20221018_132500.jpg" -- "Screen20221018_132515.jpg"[label="Nm=6, Ni=6, C=0.612245"];
"Screen20221018_132544.jpg" -- "Screen20221018_132610.jpg"[label="Nm=38, Ni=38, C=1.95876"];
"Screen20221018_132610.jpg" -- "Screen20221018_132628.jpg"[label="Nm=7, Ni=7, C=0.693069"];
"Screen20221018_132628.jpg" -- "Screen20221018_132646.jpg"[label="Nm=29, Ni=29, C=1.73653"];
"Screen20221018_132646.jpg" -- "Screen20221018_132702.jpg"[label="Nm=19, Ni=19, C=1.38686"];
"Screen20221018_132725.jpg" -- "Screen20221018_132745.jpg"[label="Nm=33, Ni=29, C=1.62011"];
"Screen20221018_132745.jpg" -- "Screen20221018_132806.jpg"[label="Nm=327, Ni=315, C=2.9689"];
"Screen20221018_132806.jpg" -- "Screen20221018_132825.jpg"[label="Nm=24, Ni=21, C=1.38158"];
"Screen20221018_132825.jpg" -- "Screen20221018_132839.jpg"[label="Nm=28, Ni=28, C=1.70732"];
"Screen20221018_132839.jpg" -- "Screen20221018_132911.jpg"[label="Nm=40, Ni=40, C=2"];
"Screen20221018_132925.jpg" -- "Screen20221018_132958.jpg"[label="Nm=24, Ni=24, C=1.57895"];
"Screen20221018_132958.jpg" -- "Screen20221018_133013.jpg"[label="Nm=41, Ni=39, C=1.92118"];
"Screen20221018_133013.jpg" -- "Screen20221018_133031.jpg"[label="Nm=22, Ni=21, C=1.43836"];
"Screen20221018_133031.jpg" -- "Screen20221018_133043.jpg"[label="Nm=39, Ni=39, C=1.9797"];
"Screen20221018_133043.jpg" -- "Screen20221018_133057.jpg"[label="Nm=8, Ni=8, C=0.769231"];
"Screen20221018_133218.jpg" -- "Screen20221018_133240.jpg"[label="Nm=41, Ni=41, C=2.0197"];
"Screen20221018_133240.jpg" -- "Screen20221018_133326.jpg"[label="Nm=28, Ni=24, C=1.46341"];
"Screen20221018_133326.jpg" -- "Screen20221018_133343.jpg"[label="Nm=40, Ni=34, C=1.7"];
"Screen20221018_133343.jpg" -- "Screen20221018_133352.jpg"[label="Nm=10, Ni=10, C=0.909091"];
"Screen20221018_133352.jpg" -- "Screen20221018_133417.jpg"[label="Nm=25, Ni=25, C=1.6129"];
"Screen20221018_133417.jpg" -- "Screen20221018_133438.jpg"[label="Nm=11, Ni=11, C=0.973451"];
"Screen20221018_133438.jpg" -- "Screen20221018_133453.jpg"[label="Nm=9, Ni=9, C=0.841121"];
"Screen20221018_133506.jpg" -- "Screen20221018_133519.jpg"[label="Nm=12, Ni=12, C=1.03448"];
"Screen20221018_133519.jpg" -- "Screen20221018_133536.jpg"[label="Nm=77, Ni=77, C=2.47588"];
"Screen20221018_133536.jpg" -- "Screen20221018_133558.jpg"[label="Nm=8, Ni=8, C=0.769231"];
"Screen20221018_133558.jpg" -- "Screen20221018_133613.jpg"[label="Nm=28, Ni=28, C=1.70732"];
"Screen20221018_133613.jpg" -- "Screen20221018_133623.jpg"[label="Nm=15, Ni=15, C=1.2"];
"Screen20221018_133640.jpg" -- "Screen20221018_133650.jpg"[label="Nm=61, Ni=49, C=1.86312"];
"Screen20221018_133132.jpg";
"Screen20221018_133631.jpg";
}
```

only two images are not processed, I havn't analized why. For a graphic representation visit [graphviz](https://dreampuf.github.io/GraphvizOnline)

![result](https://user-images.githubusercontent.com/32765578/196996162-4e33a7ab-a74d-4327-9e6b-0887540d7011.jpg)

Here is a low resolution result of the stitching

**Comment by emard** [comment]
HI

Thanks for the improvement! I have tried options you recommended
for other similar images done with 2x less magnification and it stitched
them all, compared to my trying when only half of images stitched.

I can experiment with more overlapping, of course if you find improvement
on the difficult set, I'd like to try it

Regards, Davor

**Comment by lukasalexanderweber** [comment]
You could try the other two detectors brisk or akaze, but other than that I dont see that much potential for further optimization but more overlap. Cheers Lukas
